instance.tf

```diff
@@ -19,7 +19,7 @@ resource "google_sql_database_instance" "instance" {
       enabled                        = coalesce(var.backup_config.enabled, var.highly_available)
       start_time                     = var.backup_config.start_time
       point_in_time_recovery_enabled = var.backup_config.point_in_time_recovery_enabled
-      location                       = var.backup_config.enabled == true ? var.backup_config.location : null
+      location                       = coalesce(var.backup_config.enabled, var.highly_available) ? var.backup_config.location : null
```

Comment:
Should probably shove the `coalesce(...)` into a local or smth. But I noticed some more "inefficient" logic. Something for a future iteration to take care of all that.